### PR TITLE
103585 Director config refactor

### PR DESCRIPTION
**Pull request by dawidkeshop**

### CODE REVIEW

File: src/CaptainHook.DirectorService/Program.cs [hunk 1]
```diff
@@ -16,46 +17,56 @@
...
         /// </summary>
         private static async Task Main()
         {
             try
             {
-                var appSettings = TempConfigLoader.Load();
+                var environmentName = EswDevOpsSdk.GetEnvironmentName();
+
+                var configuration = new ConfigurationBuilder()
+                    .UseDefaultConfigs(environment: environmentName)
```
Comment (tabman83) [inline src/CaptainHook.DirectorService/Program.cs:38]: We don't need to pass the environment name explicitly here
https://github.com/eShopWorld/devops/blob/1cc8623bb993c157dd3fb740a1cb2da40727295c/src/Eshopworld.DevOps/Extensions/ConfigurationExtensions.cs#L164

File: src/CaptainHook.DirectorService/appsettings.CI.json [hunk 2]
```diff
@@ -0,0 +1,3 @@
+{
+  "KeyVaultInstanceName": "esw-platform-ci-we"
```
Comment (tabman83) [inline src/CaptainHook.DirectorService/appsettings.CI.json:2]: We don't need this here - key will be inferred by the KEYVAULT_URL environment variable on each environment.
The Service Bus Subscription Id should be stored here instead as it is not a secret.

File: src/CaptainHook.DirectorService/appsettings.PREP.json [hunk 3]
```diff
@@ -0,0 +1,3 @@
+{
+  "KeyVaultInstanceName": "esw-platform-prep-we"
```
Comment (tabman83) [inline src/CaptainHook.DirectorService/appsettings.PREP.json:2]: We don't need this here - key will be inferred by the KEYVAULT_URL environment variable on each environment.
The Service Bus Subscription Id should be stored here instead as it is not a secret.

File: src/CaptainHook.DirectorService/appsettings.SAND.json [hunk 4]
```diff
@@ -0,0 +1,3 @@
+{
+  "KeyVaultInstanceName": "esw-platform-sand-we"
```
Comment (tabman83) [inline src/CaptainHook.DirectorService/appsettings.SAND.json:2]: We don't need this here - key will be inferred by the KEYVAULT_URL environment variable on each environment.
The Service Bus Subscription Id should be stored here instead as it is not a secret.

File: src/CaptainHook.DirectorService/appsettings.TEST.json [hunk 5]
```diff
@@ -0,0 +1,3 @@
+{
+  "KeyVaultInstanceName": "esw-platform-test-we"
```
Comment (tabman83) [inline src/CaptainHook.DirectorService/appsettings.TEST.json:2]: We don't need this here - key will be inferred by the KEYVAULT_URL environment variable on each environment.
The Service Bus Subscription Id should be stored here instead as it is not a secret.

File: src/CaptainHook.DirectorService/appsettings.Development.json [hunk 6]
```diff
@@ -0,0 +1,3 @@
+{
+  "KeyVaultInstanceName": "esw-platform-ci-we"
```
Comment (tabman83) [inline src/CaptainHook.DirectorService/appsettings.Development.json:2]: Ok to have this here - the user might not have the KEYVAULT_URL environment variable defined.

However the Service Bus Subscription Id should be stored here as it is not a secret.

**Review by tabman83 — CHANGES_REQUESTED** [review]

File: src/CaptainHook.DirectorService/Program.cs (hunk 1 above)
Comment (dawidkeshop) [reply src/CaptainHook.DirectorService/Program.cs]: Good catch, thanks!

File: src/CaptainHook.DirectorService/appsettings.CI.json (hunk 2 above)
Comment (dawidkeshop) [reply src/CaptainHook.DirectorService/appsettings.CI.json]: done

File: src/CaptainHook.DirectorService/appsettings.PREP.json (hunk 3 above)
Comment (dawidkeshop) [reply src/CaptainHook.DirectorService/appsettings.PREP.json]: done

File: src/CaptainHook.DirectorService/appsettings.SAND.json (hunk 4 above)
Comment (dawidkeshop) [reply src/CaptainHook.DirectorService/appsettings.SAND.json]: done

File: src/CaptainHook.DirectorService/appsettings.TEST.json (hunk 5 above)
Comment (dawidkeshop) [reply src/CaptainHook.DirectorService/appsettings.TEST.json]: done

File: src/CaptainHook.DirectorService/appsettings.Development.json (hunk 6 above)
Comment (dawidkeshop) [reply src/CaptainHook.DirectorService/appsettings.Development.json]: done

**Review by tabman83 — APPROVED** [review]

**Pull request merged**